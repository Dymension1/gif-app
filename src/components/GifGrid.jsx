import PropTypes from 'prop-types';

import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

//import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs( category );

  return (
    <>
      <h1>{ category }</h1>
      {
        isLoading && ( <h2>Loading...</h2> )
      }
      
      <div className="card-grid">
        {
          images.map(( image ) => (
            <GifItem key={ image.id }
            { ...image }
            />
          ))
        }
      </div>
    </>
  );
};


GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}