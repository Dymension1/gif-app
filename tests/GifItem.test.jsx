import { GifItem } from "../src/components/GifItem";
import { render, screen } from "@testing-library/react";


describe("GifItem test", () => {
  
  const title = "Title";
  const url = "https://www.google.com/";

  test("Pruebas en GifItem ", () => {

    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();

  });


  test("debe de mostrar la imagen con URL y ALT indicado", () => {

    render(<GifItem title={title} url={url} />);

    // expect( screen. getByRole('img').src ).toBe( title );
    // expect( screen. getByRole('img').alt ).toBe( URL );

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);

  });


  test('Debe mostrar title en el componente', () => { 

    render(<GifItem title={title} url={url} />);

    expect( screen.getByText(title) ).toBeTruthy();
  });


});
