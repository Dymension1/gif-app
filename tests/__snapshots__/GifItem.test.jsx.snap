// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem test Pruebas en GifItem  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title"
      src="https://www.google.com/"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
